packages/api-client-next/src/index.ts

```diff
@@ -88,16 +91,14 @@ export function createAPIClient<
   accessToken: string;
   contextToken?: string;
   onContextChanged?: (newContextToken: string) => void;
+  defaultHeaders?: ClientHeaders;
 }) {
-  const defaultHeaders: Record<string, string> = {
+  const defaultHeaders = createHeaders({
     "sw-access-key": params.accessToken,
     Accept: "application/json",
-  };
-
-  // protection from setting "null" or "undefined" as a token in API side
-  if (params.contextToken) {
-    defaultHeaders["sw-context-token"] = params.contextToken;
-  }
+    ...(params.contextToken && { "sw-context-token": params.contextToken }),
```

Comment:
you've removed "protection from setting null or underined" above:

- what value will be sent in a very first request for `sw-context-token` when value it's not provided?
we need to ensure if it's not serialized to the empty string `""` because in that case we'd borrow a session under this context token which accepts every kind of string. maybe it's worth considering a backend issue to prevent this one edge case.